### PR TITLE
Add notes to practical support

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

Adds the ability to save notes under practical support; adjusts the structure of the practical support edit pane to include tabs for easier nav.

This pull request makes the following changes:
* Adds the ability to attach notes to practical support entries
* Adds tabs for practical supports, for easier organization

image

image

image

It relates to the following issue #s: 
* Fixes #X
* Bumps #Y

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: app/controllers/notes_controller.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Create notes associated with patients. 
 class NotesController < ApplicationController
-  before_action :find_patient, only: [:create]
+  before_action :find_object, only: [:create]
```
Comment (colinxfleming) [inline app/controllers/notes_controller.rb:3]: some generalization

File: app/views/practical_supports/_edit.html.erb [hunk 2]
```diff
@@ -1,37 +1,75 @@
...
+                          class: 'col' %>
+
+          <div class="row">
+            <div class="col info-form-left">
+              <%= f.submit t('patient.notes.submit'), class: "btn btn-primary" %>
+            </div>
+          </div>
+        <% end %>
+
+        <div class="mt-5">
```
Comment (colinxfleming) [inline app/views/practical_supports/_edit.html.erb:66]: I had a hard time getting this to overflow properly. It's not mega janky but it's definitely kinda ugly. I'm fine leaving it but if there's something easy I'm missing that'd be killer.

File: app/views/practical_supports/_edit.html.erb [hunk 3]
```diff
@@ -1,37 +1,75 @@
...
+          <%= render partial: 'notes/notes_table',
```
Comment (colinxfleming) [inline app/views/practical_supports/_edit.html.erb:67]: I was really pleasantly surprised how much of this I was able to reuse without thinking about it much.

**Review by colinxfleming — COMMENTED** [review]
notes

**Review by lomky — COMMENTED** [review]
Two thoughts!
- Display Refreshing: 
  - Good: when you update the psupport fields, the pages updates the display one on modal close with the new info
  - Bug: when you add a new note, the display does not show the new one on modal close, but only when you refresh.
- as a followup: this is the first time we have 2+ places to see notes, and I worry a little about notes getting lost and folks having to search for them. Spitballing:
    - add a "Show all Notes" button on the main patient Notes page
    - when clicked, each psupport summary line is shown with all its notes
    - the psupport summary can be clicked to access the modal

**Comment by colinxfleming** [comment]
I'll bump the refresh. I think I'm less concerned about all notes in the same pool, but willing to be wrong about it; I think for expedience let's see if that one comes up organically?

**Comment by colinxfleming** [comment]
@lomky that weird behavior around creation should be fixed in [d8505da](https://github.com/DARIAEngineering/dcaf_case_management/pull/3249/commits/d8505daac809ba088e106a6a7c2bc5ef9c85309c) (along with another mild bug around notes_log being an id not a class). If you want a consolidated notes thing I'm good with that, but we should do it as a followup, since this PR is needed to unstick some slightly time sensitive work and finish out our direct commit to the fund that asked for this.

**Comment by lomky** [comment]
totally cool with holding on that for follow up or request

**Review by lomky — APPROVED** [review]
Adding remove a note to system tests would be nice, but your discretion if you'd rather get this out to folks!

**Comment by colinxfleming** [comment]
merging as is, let's do followups as they arise!

**Pull request merged**